examples/refsi/hal_refsi/source/refsi_hal_g1.cpp

```diff
@@ -273,12 +273,14 @@ bool refsi_g1_hal_device::mem_copy(hal::hal_addr_t dst, hal::hal_addr_t src,
             dst, src, size);
   }
 
-  std::vector<uint8_t> temp(size);
-  if (!mem_read(temp.data(), src, size, locker)) {
-    return false;
-  }
-  if (!mem_write(dst, temp.data(), size, locker)) {
-    return false;
+  if (size) {
```

Comment:
I'm assuming this is because `mem_read` and `mem_write` still require `size` to be greater than zero? Are they used anywhere directly where a size of zero should be allowed?